### PR TITLE
LINK-1123 | Add code quality tools

### PR DESCRIPTION
**Pull request by charn**
This PR adds code quality related tools: black, flake8 and isort. Configuration for each of these tools is added as well. Result of running these tools for the whole code base shall follow in a separate PR.

### CODE REVIEW

**Review by danipran — CHANGES_REQUESTED** [review]
Need to account for flake8-black shenanigans. See: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8

File: setup.cfg [hunk 1]
```diff
@@ -1,4 +1,8 @@
 [flake8]
```
[the remark this answers is not in the document]
Comment (charn) [reply setup.cfg:1]: I shall add this. :+1:

File: setup.cfg [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
-extend-ignore = E741
-max-line-length=119
-exclude=migrations
+exclude = migrations,tests
+max-line-length = 120
```
[the remark this answers is not in the document]
Comment (charn) [reply setup.cfg]: black doesn't actually split long strings. So if we occasionally want to allow longer strings imho 120 is a good value in that sense. :shrug:.

**Review by danipran — APPROVED** [review]
LGTM 👍 Tested pre-commit config locally, works as well.

**Pull request merged**